src/preload/syscall_hook.S

```diff
@@ -203,14 +203,14 @@ _syscall_hook_trampoline:
         movdqa 0x70(%esp),%xmm6
         movdqa 0x80(%esp),%xmm7
 
-        pushw (saved_flags)
-        popfw
-        /* From here on, non-application flag changes are not allowed */
-
         /* Restore ESP */
         mov %ebp, %esp
         .cfi_def_cfa_register %esp
 
+        pushw (saved_flags)
+        popfw
```

Comment:
I don't think this is safe, because we're corrupting the redzone.

I think we could do what we do on the way in: set SP to point to saved_flags and then popfw, then set SP to BP. I didn't do that originally because I was worried about stack validity but I think that's actually not an issue if we do this while the CFA register is BP.

Reply:
Alright, done.